.clang-tidy

```diff
@@ -0,0 +1,11 @@
+---
+Checks:          '*,-android-cloexec-fopen,-misc-unused-parameters'
```

Comment:
> will that not turn on code-base specific checks?

It does. My thinking here:
* It's a C project. Most checks are disabled because of that anyways (e. g. all of `modernize-` is completely useless here)
* With this patch the code is clang-tidy clean
* These are Warnings and not treated as errors (see one line below, which is a whitelist, not a blacklist). New checks in clang-tidy won't result in broken builds

We can change it to something like `bugprone-*,misc-*,readability-*,performance-*` if you like as a starter. But I think it's more important to think about which checks we want to treat as errors and should fail CI (none? the ones listed below already?)

Reply:
I did exactly what I suggested in my last paragraph.

Reply:
Thanks, that set is already pretty extensive and covers the generic stuff.